remote-api/src/main/java/org/alfresco/rest/framework/resource/parameters/ListPages.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * #%L
+ * Alfresco Remote API
+ * %%
+ * Copyright (C) 2005 - 2022 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+
+package org.alfresco.rest.framework.resource.parameters;
+
+import org.alfresco.query.ListBackedPagingResults;
+import org.alfresco.query.PagingRequest;
+import org.alfresco.query.PagingResults;
+import org.alfresco.rest.api.impl.Util;
+import org.alfresco.rest.api.search.context.SearchContext;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+public class ListPages<T> extends CollectionWithPagingInfo<T>
```

Comment:
(Why) do we need to add this subclass to the REST framework ?

Is this for in-memory paging (if underlying service does not support paging - with assumption that items are listed in a repeatable order over multiple paged calls) ?

If so, I wonder if there are any other existing implemented areas that do something similar that might benefit ?

Reply:
Thanks. 

Do we have a list of "everyone" (eg. code links) that we could add to your comment (for reference) ?

Maybe also log as tech debt in Jira (ie. option to go back and rationalise others) to use your generic approach ?

Reply:
"Everyone" is not a proper wording. A lot of solutions (better sentence) handles the paging in its custom way. I don't have any list, but it's possible to do some research. Some of the solutions are old and exists there for long time.